packages/teams-js/src/private/files.ts

```diff
@@ -296,6 +296,70 @@ export namespace files {
     extension: string;
   }
 
+  export enum CloudStorageProviderFileAction {
+    New = 'NEW',
+    Rename = 'RENAME',
+    Download = 'DOWNLOAD',
+    Upload = 'UPLOAD',
+    Delete = 'DELETE',
+    Logout = 'LOGOUT',
+  }
+
+  export interface I3PCloudStorageProviderRequest<T> {
+    content: T;
+  }
+
+  /**
+   * @hidden
+   * Hide from docs
+   *
+   * Base interface for 3P cloud storage provider action request content
+   */
+  interface IBase3PCloudStorageProviderContent {
+    action: CloudStorageProviderFileAction;
+    providerCode: CloudStorageProvider;
+  }
+
+  /**
+   * @hidden
+   * Hide from docs
+   *
+   * Interface representing 3P cloud storage provider logout action
+   */
+  interface I3PCloudStorageProviderLogoutContent extends IBase3PCloudStorageProviderContent {
+    action: CloudStorageProviderFileAction.Logout;
+  }
+
+  /**
+   * @hidden
+   * Hide from docs
+   *
+   * Interface for 3P cloud storage provider action specific request content
+   */
+  interface I3PCloudStorageProviderActionContent extends IBase3PCloudStorageProviderContent {
```

Comment:
It would be great to use the type system to make multiple types for each of the actions you want to support. Looking at the code below, it seems like depending on the action type (NEW, RENAME, etc.) you are expecting different combinations of these properties to be set, and returning errors otherwise. This will make it tough for callers to understand without going through trial and error. Making types like "I3PCloudStorageProviderActionContentForNew" (or whatever you want to name it) that requires an existingFileName and then either omits or makes optional the other parameters would make it a lot easier to use this for future callers.

Reply:
Splitted the common action method to multiple action methods, to be specific create action method for New and Rename.
The updated common method still represents 3 actions for file(s):
1. Upload 2. Download 3. Delete 
The reason of these 3 actions being represented by a common type is that they have exact same input parameters.